### PR TITLE
1316 fix broken rtl cases

### LINKED ISSUE #1316: Fix broken RTL cases
We did a first sweep in #1302, but let's now go through again and fix any broken stories. Here are a few I found quickly:

Screen Shot 2020-12-10 at 2 32 44 PM
Screen Shot 2020-12-10 at 2 31 15 PM
Screen Shot 2020-12-10 at 2 30 59 PM
Screen Shot 2020-12-10 at 2 26 18 PM

### PR DESCRIPTION
**Pull request by austingardner**
## Description
Fixed several broken RTL cases. closes #1316 

squash please! 

## Context
We previously had added RTL to several stories, but in some cases the styling had become wonky. This PR fixes the cases we identified. 

## Screenshots
Screen Shot 2021-01-06 at 2 47 06 PM
Screen Shot 2021-01-06 at 2 46 08 PM
Screen Shot 2021-01-06 at 2 45 44 PM
Screen Shot 2021-01-06 at 2 45 31 PM
Screen Shot 2021-01-06 at 2 45 15 PM
Screen Shot 2021-01-06 at 2 44 49 PM

### CODE REVIEW

File: src/less/carousel/stories/carousel.stories.js [hunk 1]
```diff
@@ -131,8 +131,8 @@ export const RTL = () => `
                 <span>Top Products - Slide 1 of 4</span>
             </h4>
             <button class="carousel__control carousel__control--prev" aria-label="Previous Slide - Top Products">
-                <svg aria-hidden="true" class="icon icon--carousel-next" focusable="false">
-                    <use xlink:href="#icon-carousel-next"></use>
+                <svg aria-hidden="true" class="icon icon--carousel-prev" focusable="false">
+                    <use xlink:href="#icon-carousel-prev"></use>
```
Comment (austingardner) [inline src/less/carousel/stories/carousel.stories.js:135]: this was a correction - I had previously switched these, but that switch was in error.

**Review by ianmcburnie — CHANGES_REQUESTED** [review]
There are some lint errors when I try and run locally:

Screen Shot 2021-01-07 at 10 15 57 AM

**Comment by austingardner** [comment]
Lint errors fixed -> definitely we should add a hook that runs that before commit or push

File: src/less/toast-dialog/base/toast-dialog.less [hunk 2]
```diff
@@ -144,3 +144,9 @@ button.toast-dialog__close {
         margin: 16px 24px 24px;
     }
 }
+
+[dir="rtl"] {
+    .btn.btn--primary {
```
Comment (ianmcburnie) [inline src/less/toast-dialog/base/toast-dialog.less:149]: Nitpicking, but is the chained selector necessary? We try and avoid them if we can.

Comment (austingardner) [reply src/less/toast-dialog/base/toast-dialog.less:149]: probably not, I was just trying to find things that worked. 

I'll adjust it

File: src/less/window-notice/base/window-notice.less [hunk 3]
```diff
@@ -50,3 +50,15 @@
         width: 100%;
     }
 }
+
+[dir="rtl"] {
+    .window-notice__header {
+        display: flex;
```
Comment (ianmcburnie) [inline src/less/window-notice/base/window-notice.less:56]: We shouldn't really need to modify the display type in the RTL block. These rules may need moving to the base styles instead.

File: src/less/menu/base/menu.less [hunk 4]
```diff
@@ -82,7 +83,7 @@ div.menu__item[role^="menuitem"][aria-checked="true"] svg.icon {
 a.fake-menu__item span,
 button.fake-menu__item span,
 div.menu__item[role^="menuitem"] span {
-    flex: 1 0 auto;
+    flex: 0 0 auto;
```
Comment (ianmcburnie) [inline src/less/menu/base/menu.less:86]: Curious as to what this change was for?

Comment (austingardner) [reply src/less/menu/base/menu.less:86]: I don't know why this worked, but it totally fixed the issue.

Comment (ianmcburnie) [reply src/less/menu/base/menu.less:86]: Ok, just double check for any regressions in non RTL mode

File: src/less/window-notice/base/window-notice.less (hunk 3 above)
Comment (austingardner) [reply src/less/window-notice/base/window-notice.less:56]: I deleted the `display: flex` there and turns out it wasn't needed.

File: src/less/menu/base/menu.less (hunk 4 above)
Comment (austingardner) [reply src/less/menu/base/menu.less:86]: ✅

File: src/less/section-notice/base/section-notice.less [hunk 5]
```diff
@@ -84,3 +84,9 @@ span[role="region"].section-notice {
         width: auto;
     }
 }
+
+[dir="rtl"] {
+    .icon--confirmation-filled {
```
Comment (ianmcburnie) [inline src/less/section-notice/base/section-notice.less:89]: Add `.section-notice__header` scope here. Also should apply to all 3 types of icon

File: src/less/window-notice/base/window-notice.less [hunk 6]
```diff
@@ -50,3 +50,12 @@
         width: 100%;
     }
 }
+
```
Comment (ianmcburnie) [inline src/less/window-notice/base/window-notice.less:53]: As discussed we can remove this block entirely

File: src/less/inline-notice/base/inline-notice.less [hunk 7]
```diff
@@ -25,3 +25,9 @@ span.inline-notice {
         font-weight: bold;
     }
 }
+
+[dir="rtl"] {
+    .icon.icon--confirmation-filled {
```
Comment (ianmcburnie) [inline src/less/inline-notice/base/inline-notice.less:30]: This needs correct scope too (similar to section notice)

File: src/less/filter-menu-button/base/filter-menu-button.less [hunk 8]
```diff
@@ -263,3 +263,14 @@ button.filter-menu-button__footer {
         display: block;
     }
 }
+
+[dir="rtl"] {
+    .icon.icon--dropdown {
```
Comment (ianmcburnie) [inline src/less/filter-menu-button/base/filter-menu-button.less:268]: Needs correct scope

**Review by ianmcburnie — APPROVED** [review]

**Review by agliga — APPROVED** [review]

**Pull request merged**